### PR TITLE
Annotation content slot

### LINKED ISSUE #2403: Image annotation: support 'slot' syntax for the content attribute
For image annotations, perhaps we can support the `slot` syntax for the content attribute, to allow more complicated content?

image

### PR DESCRIPTION
**Pull request by yucheng11122017**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [X] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

Resolves #2403

**Overview of changes:**
Enable markdown for content, header and label in annotation point

**Anything you'd like to highlight/discuss:**
Per the issue, I suggested renaming 'content' to something else because it is a little misleading. However, this would result in a breaking change. So will leave it as it is for now. 

**Testing instructions:**
Try to use markdown in content, header and label in annotation point

**Proposed commit message: (wrap lines at 72 characters)**
Enable markdown for content, header and label in annotation point

---

**Checklist:** :ballot_box_with_check:

- [X] Updated the documentation for feature additions and enhancements
- [X] Added tests for bug fixes or features
- [X] Linked all related issues
- [X] No unrelated changes 

---

**Reviewer checklist:**

Indicate the [SEMVER](https://semver.org/) impact of the PR:
- [ ] Major (when you make incompatible API changes)
- [x] Minor (when you add functionality in a backward compatible manner)
- [ ] Patch (when you make backward compatible bug fixes)

At the end of the review, please label the PR with the appropriate label: `r.Major`, `r.Minor`, `r.Patch`.

Breaking change release note preparation (if applicable):
- To be included in the release note for any feature that is made obsolete/breaking

> Give a brief explanation note about:
>   - what was the old feature that was made obsolete
>   - any replacement feature (if any), and
>   - how the author should modify his website to migrate from the old feature to the replacement feature (if possible).

### CODE REVIEW

**Comment by yucheng11122017** [comment]
Been trying to add unit tests for annotation but having problems with stubbing `v-popover`. 
When I try to stub it, it simply creates an empty tag, hence all the annotation points are missing. 
Would appreciate any advice on this :)

File: packages/vue-components/src/__tests__/Annotation.spec.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { mount } from '@vue/test-utils';
+import Annotation from '../annotations/Annotate.vue';
+import AnnotationPoint from '../annotations/AnnotatePoint.vue';
+
+const DEFAULT_STUBS = {
+  'a-point': AnnotationPoint,
+  'v-popover': true,
```
Comment (EltonGohJH) [inline packages/vue-components/src/__tests__/Annotation.spec.js:7]: import { Dropdown } from 'floating-vue';

v-popover = Dropdown

File: packages/vue-components/src/__tests__/Annotation.spec.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    const wrapper = mount(Annotation, {
+      propsData: {
+        src: './annotateSampleImage.png',
+      },
+      slots: {
+        default: ANNOTATION_POINTS,
+      },
+      stubs: DEFAULT_STUBS,
+    });
+
```
Comment (EltonGohJH) [inline packages/vue-components/src/__tests__/Annotation.spec.js:48]: I tried this and this works. This waits for dom changes. (We are waiting for child components to be mounted)
`await wrapper.vm.$nextTick(); `

(Add to the rest if you are using this method!)

**Review by EltonGohJH — CHANGES_REQUESTED** [review]
Hi @yucheng11122017,

I tested the changes on my end wdyt of the requested changes?
From my observations
1. If you look at the annotatepoint, it is waiting for it to be fully mounted before rendering (waiting for isMounted flag to fully render). 
So, the solution I used wait for dom changes. It works but I think it might not be the recommended practice. (I think it is fine since it is deterministic)
You can try other methods from here (I think other methods do not really wait for child component to be mounted. So, I am not sure if there are any other methods) https://v1.test-utils.vuejs.org/guides/testing-async-components.html

2. v-postpover = dropdown from floating-vie

**Comment by EltonGohJH** [comment]
image
@yucheng11122017 
There is also a weird tooltip artefact on click. Is this a separate bug?

**Comment by yucheng11122017** [comment]
> image @yucheng11122017 There is also a weird tooltip artefact on click. Is this a separate bug?

Yup I noticed that prior to this PR - basically the popover is still present even though there is no content and header which resulted in that arrow

**Pull request merged**